### DISCUSSION TITLE
Cannot write ecu for Android Auto activation

### DISCUSSION THREAD

**Discussion by Cliomk4rs** [cedricp/ddt4all#1027] [Q&A]
Hi all need some assistance, I cannot write the changes to MFD to activate android, feeling defeated. 

Car: 2014 Clio 4 RS - R link
Does not have SD card (Don't know if relevant)

When I scan for ecus via CAN, MFD (nav) does not display and I get the following message: 
"Found ECU [Navigation] (no relevant ecu file found)" 

When I select MFD V5.4 form Clio IV data base and try write the changes to activate android auto, it does not apply the changes (stays as "not present*) 

I've tried increasing time out to max before writing. 

Any help would be appreciated.

**Comment by KarelSvo** [comment]
log files

**Reply by Cliomk4rs** [reply]
Sorry new to this I'll get the log files

**Reply by KarelSvo** [reply]
folder ddt4all/logs

**Reply by Cliomk4rs** [reply]
Hi @KarelSvo
ECU did attempt to write but got the following:

Found ECU [Navigation] (no relevant ECU file found) :DIAGVERSION [20] SUPPLIER [329] SOFT [2438] VERSION [28FC]
ECU uses SDS 10C0
Sending request :StartAfterSalesSession
Delay 50 ms
Switching to session mode 10 C0
Sending request :DataWrite.ECU
Delay 250 ms
Sending ELM request:2E 6C 1C C4
Bad ELM response : WRONG RESPONSE : NR: SubFunction Not Supported(7F2E12)
ELM response : WRONG RESPONSE : NR: SubFunction Not Supported(7F2E12)
Sending request :StartDiagnosticSession
Delay 250 ms
Switching to session mode 10 81

**Reply by Cliomk4rs** [reply]
[screens.txt](https://github.com/cedricp/ddt4all/files/14266448/screens.txt)
Here is log files

**Comment by AndyIreland** [comment]
![image](https://github.com/cedricp/ddt4all/assets/49172217/758873c3-ea0f-4301-9517-d4ac4d035031)

600ms can timeout.

when finished doing press home button on r-link 5 times or press 'hard Reset' in DDT4ALL  (DONT PRESS SOFT RESET!!!)

**Reply by Cliomk4rs** [reply]
Ill will try this thank you, the issue is when i hit write it doesn't not change any parameters
See image
![Uploading Screenshot_20240211_154744_com.huawei.himovie.overseas_edit_47987083511948.jpg…]()

**Reply by AndyIreland** [reply]
I couldnt see your screenshot, you didnt wait for it to upload fully before posting your message

**Reply by Cliomk4rs** [reply]
![Screenshot_20240211_154744_com huawei himovie overseas_edit_47987083511948](https://github.com/cedricp/ddt4all/assets/159680041/6a128e8c-f24e-4aa9-90b8-1734e7fe29d7)

**Reply by Cliomk4rs** [reply]
@AndyIreland sorry mate new on this platform, i do appreciate you going out of your way to help

**Reply by AndyIreland** [reply]
no problem . You are clicking on einstein icon? - you press on READ afterwards or refresh button?

**Reply by Cliomk4rs** [reply]
@AndyIreland
An update: ECU did attempt to write after following your steps, however now after failing to write I'm getting: Bad ELM response : WRONG RESPONSE : NR: SubFunction Not Supported(7F2E12)

I'll paste full code below:

Found ECU [Navigation] (no relevant ECU file found) :DIAGVERSION [20] SUPPLIER [329] SOFT [2438] VERSION [28FC]
ECU uses SDS 10C0
Sending request :StartAfterSalesSession
Delay 50 ms
Switching to session mode 10 C0
Sending request :DataWrite.ECU
Delay 250 ms
Sending ELM request:2E 6C 1C C4
Bad ELM response : WRONG RESPONSE : NR: SubFunction Not Supported(7F2E12)
ELM response : WRONG RESPONSE : NR: SubFunction Not Supported(7F2E12)
Sending request :StartDiagnosticSession
Delay 250 ms
Switching to session mode 10 81

**Reply by Cliomk4rs** [reply]
If it helps I've test elm on another Clio and successfully activated android auto
